### PR TITLE
Destroy koin rebirth

### PR DESCRIPTION
**Pull request by ziad-abdo**

### CODE REVIEW

File: command/destroy_koin.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+type DestroyKoinCommand struct{}
+
+func (c *DestroyKoinCommand) Run(msg string, event *CoinEvent) (string, error) {
+	re := regexp.MustCompile(`^(?i)destroy koin (?:for )?\<@(?P<to_slack_id>[0-9A-Z]+)\> (?:for)??(?P<reason>.+)`)
+	matches := re.FindStringSubmatch(event.Message)
+
+	if len(matches) < 3 {
+		return "Invalid create koin format. Expected something like `@Alex Koin destroy koin @alexk for being amazing`. See the channel details for command syntax.", nil
```
Comment (hex337) [inline command/destroy_koin.go]:
```suggestion
		return "Invalid destroy koin format. Expected something like `@Alex Koin destroy koin @alexk for being horrible`. See the channel details for command syntax.", nil
```

File: command/destroy_koin.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	// Get or create the to user
+	toUser, err := model.GetOrCreateUserBySlackID(toUserId)
+
+	if err != nil {
+		log.Fatalf("Could not find user with slack id %s: %s", toUserId, err.Error())
+		return "", err
+	}
+
+	canCreate, msg := canDestroyCoin(event.User, toUser)
```
Comment (hex337) [inline command/destroy_koin.go]:
```suggestion
	canDestroy, msg := canDestroyCoin(event.User, toUser)
```

**Comment by finack** [comment]
I will try to review later but I am pretty sure we don't want to `log.Fatal` or that will crash the running process. Need to refresh my memory there.

File: command/destroy_koin.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	if !canCreate {
```
Comment (hex337) [inline command/destroy_koin.go]:
```suggestion
	if !canDestroy {
```

File: command/destroy_koin.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		return msg, nil
+	}
+
+	koin, err := toUser.GetCoin()
```
Comment (hex337) [inline command/destroy_koin.go]:
```suggestion
	coin, err := toUser.GetCoin()
```

File: command/destroy_koin.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+	if err != nil {
+		log.Fatalf("Failed to Destroy Koin. Err: %s", err)
+		return fmt.Sprintf("There was a failure in the system. Koin not destroyed"), nil
+	}
+
+	err = koin.DestroyKoin()
```
Comment (hex337) [inline command/destroy_koin.go]:
```suggestion
	err = coin.DestroyKoin()
```

File: command/destroy_koin.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Rules:
+ *  Only Admin and Lords can destroy Koin
+ */
+ func canDestroyCoin(sender *model.User, receiver *model.User) (bool, string) {
+	role := sender.Role()
+
+	if receiver.GetBalance() == 0 {
+		return false, "Have pitty on this poor soul, for they have no coin left to destroy"
```
Comment (hex337) [inline command/destroy_koin.go]:
```suggestion
		return false, "Have pitty on this poor soul, for they've no coin left to destroy"
```

**Review by hex337 — DISMISSED** [review]
Copy pasta issues.

File: command/destroy_koin.go (hunk 5 above)
Comment (ziad-abdo) [reply command/destroy_koin.go]: yo, plz explain to me why our command is `create koin` but we call it `coin` in the app :|

Comment (finack) [reply command/destroy_koin.go]: koin is a localization.

Comment (ziad-abdo) [reply command/destroy_koin.go]: for slack? Can we please add support for this to be more flexible XD like https://github.com/ruby-i18n/i18n

File: command/destroy_koin.go (hunk 2 above)
Comment (ziad-abdo) [reply command/destroy_koin.go]: :( its like i'm naked.

**Comment by ziad-abdo** [comment]
@finack i hate to admit it but i did pretty much just copy pasta from here https://github.com/hex337/alex-koin-go/blob/main/command/create_coin.go#L35

So I think it should be fine? unless this has been killing the app unknowingly :D happy to change it!

**Review by finack — APPROVED** [review]
If it was good enough for Alex it is good enough for me. :shipit:

File: command/process.go [hunk 7]
```diff
@@ -67,6 +67,7 @@ func parseCommandName(msg string) (string, error) {
 		"what_am_i":   `(?i)^[[:space:]]*what[[:space:]]+am[[:space:]]+i.*`,
 		"create_coin": `(?i)^[[:space:]]*create[[:space:]]+koin.*`,
 		"stats":       `(?i)^[[:space:]]*stats$`,
+		"destroy_koin": `(?i)^[[:space:]]*destroy[[:space:]]+koin.*`,
```
Comment (hex337) [inline command/process.go:70]: nit: This isn't formatted correctly.

**Review by hex337 — APPROVED** [review]
Let the games, begin.

**Pull request merged**